markdown/guide/testing.md

```diff
@@ -1,3 +1,163 @@
+
+`engineResolverFor` gets the resolver class used by an Engine and creates an instance to be used with test modules
+
+```js
+// tests/<integration or unit>/…
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+```
+
+### Unit/Integration Testing
+
+
+To test candidates for unit/integration (e.g. components, services and controllers) declared inside an engine, you need to set a custom resolver with the engine's prefix using `engineResolverFor` helper.
+
+Assuming that we have an `in-repo-engine or engine as addon` called `admin-app` and it has a component `hello-name` and service `some-thing`. The following would be the setup to test such component from the `<app-name>` (host app or dummy app):
+
+#### Unit Testing Basics
+
+Supposing that we have in the engine a service that has a `computedFoo` computed property based on a `foo` property.
+
+```js
+// admin-app/services/some-thing-test.js
+import Service from '@ember/service';
+import { computed } from '@ember/object';
+
+export default Service.extend({
+  foo: 'bar',
+
+  computedFoo: computed('foo', function() {
+    return `computed ${this.foo}`;
+  })
+});
+```
+
+The unit test will be like this:
+
+```js
+// <app-name>/tests/unit/services/some-thing-test.js
+
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+
+const modulePrefix = 'admin-app';
+const resolver = engineResolverFor(modulePrefix);
+
+module('Unit | Service | some thing', function(hooks) {
+  setupTest(hooks, { resolver });
+
+  test('should correctly concat foo', function(assert) {
+    const someThing = this.owner.lookup('service:some-thing');
+    someThing.set('foo', 'baz');
+
+    assert.equal(someThing.get('computedFoo'), 'computed baz');
+  });
+});
+```
+
+#### Testing Components
+
+Supposing that we have in the engine this component
+
+```hbs
+{{!--  admin-app/addon/components/hello-name.hbs --}}
+Hello, {{name}}!
+```
+
+The integration test will be like this:
+
+```js
+// <app-name>/tests/integration/components/hello-name-test.js
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+
+const modulePrefix = 'admin-app';
+const resolver = engineResolverFor(modulePrefix);
+
+module('Integration | Component | hello-name', function(hooks) {
+  setupRenderingTest(hooks, { resolver });
+
+  test('it renders', async function(assert) {
+
+    await render(hbs`{{#hello-name name="Tom"}}{{/hello-name}}`);
+
+    assert.equal(this.element.textContent.trim(), 'Hello, Tom!');
+  });
+});
+```
+
+**Note: you could create a helper and then use it like `Resolver from ../helpers/admin-app/resolver`**
```

Comment:
I'm not sure what you mean by this note. Can you add some more detail here?

Reply:
fixed